File: lib-base/src/main/java/com/llj/lib/base/widget/LoadingDialog.java
package com.llj.lib.base.widget;

import android.content.Context;
import android.os.Bundle;
import android.view.Gravity;

import com.facebook.stetho.common.LogUtil;
import com.llj.lib.base.BaseDialog;
import com.llj.lib.base.R;
import com.llj.lib.net.observer.ITag;
import com.llj.lib.utils.ADisplayUtils;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public class LoadingDialog extends BaseDialog implements ITag {
    private int mTag = -1;

    public LoadingDialog(Context context) {
        super(context, R.style.no_dim_dialog);
    }

    @Override
    public int layoutId() {
        return R.layout.dialog_loading;
    }

    @Override
    public void initViews() {

    }

    @Override
    protected void setWindowParam() {
        int width = ADisplayUtils.dp2px(getContext(), 120);
        setWindowParams(width, width, Gravity.CENTER);
        setCanceledOnTouchOutside(false);
    }

    @Override
    protected void performCreate(Bundle savedInstanceState) {
        super.performCreate(savedInstanceState);

        if (mTag <= 0) {
            throw new RuntimeException("请先添加mTag");
        }

        setOnCancelListener(dialog -> {
            LogUtil.i(TAG_LOG, "cancelOkHttpCall:" + getRequestTag());
            cancelOkHttpCall(getRequestTag());
        });
    }

    @Override
    public boolean needLoadingDialog() {
        return false;
    }


    @Override
    public void setRequestTag(int tag) {
        mTag = tag;
    }

    @Override
    public int getRequestTag() {
        return mTag;
    }

    @Override
    public void showLoadingDialog() {
        show();
    }

    @Override
    public void dismissLoadingDialog() {
        dismiss();
    }
}

File: lib-base/src/main/java/com/llj/lib/base/widget/LoadingDialog.kt
package com.llj.lib.base.widget

import android.content.Context
import android.content.DialogInterface
import android.view.Gravity
import com.llj.lib.base.BaseDialog
import com.llj.lib.base.ITask
import com.llj.lib.base.R
import com.llj.lib.utils.ADisplayUtils
import java.lang.ref.WeakReference

/**
 * ArchitectureDemo
 *
 * describe:加载框
 * @author llj
 * @date 2018/5/24
 */
class LoadingDialog : BaseDialog {

  constructor(context: Context) : super(context, R.style.no_dim_dialog) {
    this.mTaskId = -1
  }

  private var mTaskId: Int


  override fun layoutId(): Int {
    return R.layout.dialog_loading
  }

  override fun initViews() {
  }


  override fun setWindowParam() {
    val width = ADisplayUtils.dp2px(context, 120f)
    setWindowParams(width, width, Gravity.CENTER)
    setCanceledOnTouchOutside(false)
  }


  override fun needLoadingDialog(): Boolean {
    return false
  }


  override fun setRequestId(taskId: Int) {
    mTaskId = taskId
  }

  override fun getRequestId(): Int {
    return mTaskId
  }

  override fun onLoadingDialogCancel(dialog: DialogInterface?) {
  }

  override fun onLoadingDialogDismiss(dialog: DialogInterface?) {
  }

  //需要重写
  override fun showLoadingDialog() {
    show()
  }

  //需要重写
  override fun dismissLoadingDialog() {
    dismiss()
  }


}
